compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -490,6 +507,17 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
             abi = Abi::Uninhabited;
         }
 
+        // If `-Z randomize-layout` is set then we increase the offset
+        // of each field by the struct's overall alignment, this ensures
+        // that everything stays well-aligned while still providing
+        // a greater element of randomness to the overall struct's and
+        // each individual field's layout
+        if repr.can_randomize_struct_fields() {
+            for offset in offsets.iter_mut() {
+                *offset += Size::from_bytes(align.pref.bytes());
+            }
+        }
```

Comment:
This only affects code that doesn't use `#[repr(C)]` but expects a certain order - that code is already broken by the sorting we do, a lot of the time, and the shuffling is much better.

It's deterministic on the definition of the type so it doesn't affect at all the case where the same definition was copy-pasted, and the layout was assumed identical.